mlflow/server/js/src/utils/KeyFilter.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * A parsed filter used to represent both the parameter and metric key filters in the UI.
+ * We currently represent each filter as a list of keys to include, in the order the user wants to
+ * display them, and with the empty list meaning "include all keys in sorted order". However, we
+ * might want to switch to a more sophisticated filtering language in the future, such as allowing
+ * a list of wildcard expressions. In any case, the apply method applies the filter to a list of
+ * keys and returns the passing ones in the order we want to display them.
+ */
+export default class KeyFilter {
```

Comment:
Would be nice to add in the comment that this class must be immutable since it's a part of a component's state.